server.js

```diff
@@ -1,71 +1,53 @@
 'use strict';
 
-require('dotenv').config({silent: true});
-
-const isDevOrTest = (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test');
-const dataDir = (process.env.DATA_DIR || './test/data');
+const fs = require('fs');
 const chalk = require('chalk');
 const _ = require('lodash');
 const Promise = require('bluebird');
-const oddcast = require('oddcast');
 const boom = require('boom');
 const express = require('express');
 
+const config = {};
+
+if (fs.existsSync('./config.js')) {
+	const userConfig = require('./config');
+	_.defaultsDeep(config, userConfig);
+} else {
+	console.log(chalk.black.bgRed('./config.js NOT FOUND'));
+	console.log(chalk.red('Loading default server configuration.'));
+	console.log(chalk.red('You may override defaults by creating your own ./config.js file like so:'));
+	console.log(chalk.red('$ cp ./default-config.js ./config.js'));
+	const defaultConfig = require('./config-default');
+	_.defaultsDeep(config, defaultConfig);
+}
+
 const middleware = require('./middleware');
 
+const oddcast = require('oddcast');
 const bus = oddcast.bus();
 const app = express();
 
-const redis = (isDevOrTest) ? require('fakeredis').createClient() : require('redis').createClient(process.env.REDIS_URI);
-
-bus.events.use({}, oddcast.inprocessTransport());
-bus.commands.use({}, oddcast.inprocessTransport());
-bus.requests.use({}, oddcast.inprocessTransport());
+// Initialize oddcast for events, commands, requests
+bus.events.use(config.oddcast.events.options, config.oddcast.events.transport);
+bus.commands.use(config.oddcast.commands.options, config.oddcast.commands.transport);
+bus.requests.use(config.oddcast.requests.options, config.oddcast.requests.transport);
 
-// Set up the store and services you want to use
-const memoryStore = require('./stores/memory');
-const redisStore = require('./stores/redis');
-const redisSearchStore = require('./stores/redis-search');
-const identityService = require('./services/identity');
-const catalogService = require('./services/catalog');
-const eventsService = require('./services/events');
-const jsonAPIService = require('./services/json-api');
-// const authorizationService = require('./services/authorization');
-// const eventsService = require('./services/events');
+function initializer(obj) {
+	console.log(chalk.black.bgBlue(`Initializing service: ${obj.service.name}`));
+	return obj.service.initialize(bus, obj.options);
+}
 
 module.exports = Promise
-	// Initialize your stores
-	.join(
-		memoryStore.initialize(bus, {types: ['platform', 'channel']}),
-		redisStore.initialize(bus, {redis, types: ['collection', 'promotion', 'video', 'view']}),
-		redisSearchStore.initialize(bus, {redis, types: ['collection', 'video']})
-	)
-
-	// Initialize your services
+	// Initialize stores
+	.all(_.map(config.stores, initializer))
```

Comment:
Actually we could have the `services/utils.js` do this mapping/loading as well and return a Promise.